python/python/tests/test_pubsub.py

```diff
@@ -0,0 +1,1800 @@
+    method,
+    client: TRedisClient,
+):
+    if method == MethodTesting.Async:
+        # assert there are no messages to read
+        with pytest.raises(asyncio.TimeoutError):
+            await asyncio.wait_for(client.get_pubsub_message(), timeout=3)
+    elif method == MethodTesting.Sync:
+        assert client.try_get_pubsub_message() is None
+        # check callback
+
+
+def create_channels_and_patterns(
+    cluster_mode, modes_and_channels: Dict[PubSubChannelModes, Set[str]]
+):
+    result: Dict[
+        Union[
+            RedisClientConfiguration.PubSubChannelModes,
+            ClusterClientConfiguration.PubSubChannelModes,
+        ],
+        Set[str],
+    ] = {}
+    for mode, channel in modes_and_channels.items():
+        if mode == PubSubChannelModes.Exact:
+            if cluster_mode:
+                result[ClusterClientConfiguration.PubSubChannelModes.Exact] = channel
+            else:
+                result[RedisClientConfiguration.PubSubChannelModes.Exact] = channel
+        elif mode == PubSubChannelModes.Pattern:
+            if cluster_mode:
+                result[ClusterClientConfiguration.PubSubChannelModes.Pattern] = channel
+            else:
+                result[RedisClientConfiguration.PubSubChannelModes.Pattern] = channel
+        else:
+            result[ClusterClientConfiguration.PubSubChannelModes.Sharded] = channel
+    return result
+
+
+def create_pubsub_subscription(
+    cluster_mode,
+    channels_and_patterns: Dict[PubSubChannelModes, Set[str]],
+    callback=None,
+    context=None,
+):
+    channels = create_channels_and_patterns(cluster_mode, channels_and_patterns)
+    if cluster_mode:
+        return ClusterClientConfiguration.PubSubSubscriptions(
+            channels_and_patterns=channels,
+            callback=callback,
+            context=context,
+        )
+    return RedisClientConfiguration.PubSubSubscriptions(
+        channels_and_patterns=channels, callback=callback, context=context
+    )
+
+
+def new_message(
+    msg: Union[RedisClient.PubSubMsg, RedisClusterClient.PubSubMsg], context: Any
+):
+    received_messages: List[RedisClient.PubSubMsg] = context
+    received_messages.append(msg)
+
+
+@pytest.mark.asyncio
+class TestPubSub:
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize(
+        "method", [MethodTesting.Async, MethodTesting.Sync, MethodTesting.Callback]
+    )
+    async def test_pubsub_basic_happy_path(
+        self,
+        request,
+        cluster_mode: bool,
+        method: MethodTesting,
+    ):
+        """
+        Tests the basic happy path for Pub/Sub functionality.
+
+        This test covers the basic pub/sub flow using three different methods:
+        Async, Sync, and Callback. It verifies that a message published to a
+        specific channel is correctly received by a subscriber.
+        """
+        channel = get_random_string(10)
+        message = get_random_string(5)
```

Comment:
Only one message? Try few, accumulate them before reading.
Use smaller sleep, 1-10 msec should be enough.